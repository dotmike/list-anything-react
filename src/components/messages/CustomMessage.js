import React from "react";
import PropTypes from "prop-types";
import { Message } from "semantic-ui-react";

const COLORS = [
  "red",
  "orange",
  "yellow",
  "olive",
  "green",
  "teal",
  "blue",
  "violet",
  "purple",
  "pink",
  "brown",
  "grey",
  "black",
  "none"
];

const TYPES_OBJ = {
  positive: { positive: true },
  success: { success: true },
  negative: { negative: true },
  error: { error: true },
  info: { info: true },
  warning: { warning: true },
  none: {}
};

const TYPES = Object.keys(TYPES_OBJ);

const CustomMessage = ({ header, content, color, type }) => {
  const userColor = color !== "none" ? { color } : {};
  return (
    <Message
      {...userColor}
      {...TYPES_OBJ[type]}
      style={{ textAlign: "center" }}
    >
      {!!header && <Message.Header>{header}</Message.Header>}
      {!!content && <Message.Content>{content}</Message.Content>}
    </Message>
  );
};

CustomMessage.propTypes = {
  // ownProps
  header: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  color: PropTypes.oneOf(COLORS),
  type: PropTypes.oneOf(TYPES)
};

CustomMessage.defaultProps = {
  header: "",
  content: "",
  color: "none",
  type: "none"
};

export default CustomMessage;
