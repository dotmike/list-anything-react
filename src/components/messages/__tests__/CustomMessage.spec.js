import CustomMessage from "../CustomMessage";

const setup = (propOverrides = {}) => {
  const props = {
    ...propOverrides
  };

  return {
    props,
    wrapperShallow: wrapperShallow(CustomMessage, props)
  };
};

const testData = {
  header: "Test",
  content: "Testando...",
  color: "red",
  type: "error"
};

describe("CustomMessage", () => {
  it("renders correctly when no prop is passed", () => {
    const { wrapperShallow: wrapper } = setup();
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when all props are passed", () => {
    const { wrapperShallow: wrapper } = setup(testData);
    expect(wrapper()).toMatchSnapshot();
  });
});
