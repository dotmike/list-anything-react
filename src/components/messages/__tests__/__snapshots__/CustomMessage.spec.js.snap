// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMessage renders correctly when all props are passed 1`] = `
<Message
  color="red"
  error={true}
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <MessageHeader>
    Test
  </MessageHeader>
  <MessageContent>
    Testando...
  </MessageContent>
</Message>
`;

exports[`CustomMessage renders correctly when no prop is passed 1`] = `
<Message
  style={
    Object {
      "textAlign": "center",
    }
  }
/>
`;
